backend/src/packages/users/user.repository.ts

```diff
@@ -88,9 +91,17 @@ class UserRepository implements Repository {
       createdAt: new Date(user.createdAt),
       updatedAt: new Date(user.updatedAt),
       fullName: user.details?.fullName ?? '',
+      isSurveyCompleted: user.details?.isSurveyCompleted ?? false,
     });
   }
 
+  public async updateIsSurveyCompleted(id: number): Promise<void> {
+    const user = await this.userModel.query().findById(id);
+    await user
+      ?.$relatedQuery(UsersRelation.DETAILS)
+      .patch({ isSurveyCompleted: true });
+  }
```

Comment:
let's pass entity here, also we need to define value for isSurveyCompleted in service
```suggestion
  public async updateIsSurveyCompleted(entity: UserEntity): Promise<void> {
    const { id, isSurveyCompleted } = entity.toObject();
  
    const user = await this.userModel.query().findById(id);
    await user
      .relatedQuery(UsersRelation.DETAILS)
      .for(id)
      .patch({ isSurveyCompleted });
  }
```

Reply:
@Liza-Veis  Could, you, please, help me to understand it? I rewrote updateIsSurveyCompleted in userRepository as you asked, and now I have tried to rewrite completeSurvey in userService to pass it to userRepository. Still, I am stuck on creating user entity, because I have this issue: 
`Argument of type '{ isSurveyCompleted: true; }' is not assignable to parameter of type '{ id: number; email: string; fullName: string; passwordHash: string; passwordSalt: string; createdAt: Date; updatedAt: Date; isSurveyCompleted: boolean; }'.
  Type '{ isSurveyCompleted: true; }' is missing the following properties from type '{ id: number; email: string; fullName: string; passwordHash: string; passwordSalt: string; createdAt: Date; updatedAt: Date; isSurveyCompleted: boolean; }': id, email, fullName, passwordHash, and 3 more.ts(2345)`
  So, I don't know how to get this entity in user service to pass it to userRepository.
![image](https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/102380564/0ec4eacc-c30a-4786-8c5f-7a4b6d1cf7ed)
![image](https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/102380564/13a944ae-a7c9-42dd-bf29-82b668777247)

Reply:
Did it in another way)